Sources/Build/BuildOperation.swift

```diff
@@ -35,28 +35,141 @@ import DriverSupport
+        outputStream: OutputByteStream,
+        observabilityScope: ObservabilityScope
+    ) {
+        self.toolsBuildParameters = toolsBuildParameters
+        self.destinationBuildParameters = destinationBuildParameters
+        self.scratchDirectory = scratchDirectory
+        self.traitConfiguration = traitConfiguration
+        self.manifestPath = manifestPath ?? destinationBuildParameters.llbuildManifest
+        self.databasePath = databasePath ?? scratchDirectory.appending("build.db")
+        self.buildDescriptionPath = buildDescriptionPath ?? destinationBuildParameters.buildDescriptionPath
+        self.fileSystem = fileSystem
+        self.logLevel = logLevel
+        self.outputStream = outputStream
+        self.observabilityScope = observabilityScope
+    }
+
+    func buildParameters(for destination: BuildParameters.Destination) -> BuildParameters {
+        switch destination {
+        case .host: self.toolsBuildParameters
+        case .target: self.destinationBuildParameters
+        }
+    }
+
+    func buildEnvironment(for destination:  BuildParameters.Destination) -> BuildEnvironment {
+        switch destination {
+        case .host: self.toolsBuildParameters.buildEnvironment
+        case .target: self.destinationBuildParameters.buildEnvironment
+        }
+    }
+
+    func shouldSkipBuilding(for destination: BuildParameters.Destination) -> Bool {
+        switch destination {
+        case .host: self.toolsBuildParameters.shouldSkipBuilding
+        case .target: self.destinationBuildParameters.shouldSkipBuilding
+        }
+    }
+
+    func toolchain(for description: BuildParameters.Destination) -> any PackageModel.Toolchain {
+        switch description {
+        case .host: self.toolsBuildParameters.toolchain
+        case .target: self.destinationBuildParameters.toolchain
+        }
+    }
+
+    func buildPath(for description: BuildParameters.Destination) -> AbsolutePath {
+        switch description {
+        case .host: self.toolsBuildParameters.buildPath
+        case .target: self.destinationBuildParameters.buildPath
+        }
+    }
+
+    func dataPath(for description: BuildParameters.Destination) -> AbsolutePath {
+        switch description {
+        case .host: self.toolsBuildParameters.dataPath
+        case .target: self.destinationBuildParameters.dataPath
+        }
+    }
+
+    func buildDescriptionPath(for description: BuildParameters.Destination) -> AbsolutePath {
+        switch description {
+        case .host: self.toolsBuildParameters.buildDescriptionPath
+        case .target: self.destinationBuildParameters.buildDescriptionPath
+        }
+    }
+
+    func configuration(for destination: BuildParameters.Destination) -> BuildConfiguration {
+        switch destination {
+        case .host: self.toolsBuildParameters.configuration
+        case .target: self.destinationBuildParameters.configuration
+        }
+    }
+
+    static func based(
```

Comment:
As this is a value type, why not mark `manifestPath`, `databasePath`, and `buildDescriptionPath` with `package` access and avoid this `static func` altogether? Not sure it's worth maintaining a separate function for straightforward updating a few properties on a value type, unless we have bigger plans for it in the future.